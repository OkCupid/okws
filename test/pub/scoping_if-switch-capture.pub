{%
 // If and switch do not have their own scopes...

 locals {x:"top_of_file\n"}

 def print_captured() {
     print (x);
 }

 print (x);
 print_captured();

 if (true) {
     locals {x:"if\n"};
 }

 print(x);         // <------------ if
 print_captured(); // <------- top_of_file

 switch (10) {
 case (10) {locals{x:"switch_case\n"}}
 }
 print(x); // <--------- switch_case
%}
