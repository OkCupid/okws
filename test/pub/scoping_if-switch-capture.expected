top_of_file
top_of_file
if
top_of_file
switch_case

